src/utils/read_all.c

```diff
@@ -11,13 +11,22 @@
 #if HAVE_CONFIG_H
 #include "config.h"
 #endif  // HAVE_CONFIG_H
-
+#define _GNU_SOURCE
 #include <errno.h>
 #include <stdlib.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <unistd.h>
-
+#include <fcntl.h>
+#include <sys/mman.h>
 #include "read_all.h"
 
+void * allocate_aligned(size_t size) {
+  size_t alignment = getpagesize();
+  int malloc_offset = (size + alignment) % alignment; 
+  void* mem = malloc(size + alignment);
+  return (char*) mem + malloc_offset;
+}
```

Comment:
I do not think that this will guarantee that the address is a multiple of alignment, while the size of the usable block is. In addition, i am wondering if it would be safe to use such an address with munmap().